google-cloud-datastore/src/main/java/com/google/cloud/datastore/DatastoreImpl.java

```diff
@@ -337,33 +360,11 @@ public Iterator<Entity> get(Iterable<Key> keys, ReadOption... options) {
     return get(toReadOptionsPb(options), Iterables.toArray(keys, Key.class));
   }
 
-  private static com.google.datastore.v1.ReadOptions toReadOptionsPb(ReadOption... options) {
-    com.google.datastore.v1.ReadOptions readOptionsPb = null;
+  private com.google.datastore.v1.ReadOptions toReadOptionsPb(ReadOption... options) {
     if (options != null) {
-      Map<Class<? extends ReadOption>, ReadOption> optionsByType =
-          ReadOption.asImmutableMap(options);
-
-      if (optionsByType.containsKey(EventualConsistency.class)
-          && optionsByType.containsKey(ReadTime.class)) {
-        throw DatastoreException.throwInvalidRequest(
-            "Can not use eventual consistency read with read time.");
-      }
-
-      if (optionsByType.containsKey(EventualConsistency.class)) {
-        readOptionsPb =
-            com.google.datastore.v1.ReadOptions.newBuilder()
-                .setReadConsistency(ReadConsistency.EVENTUAL)
-                .build();
-      }
-
-      if (optionsByType.containsKey(ReadTime.class)) {
-        readOptionsPb =
-            com.google.datastore.v1.ReadOptions.newBuilder()
-                .setReadTime(((ReadTime) optionsByType.get(ReadTime.class)).time().toProto())
-                .build();
-      }
+      return this.readOptionProtoPreparer.prepare(Arrays.asList(options));
     }
-    return readOptionsPb;
+    return null;
```

Comment:
Hi @kolea2 , I've maintained the existing contract of this method, that is to return null when no `ReadOptions` are provided by the user.
Since  we are using Java 8, how about if we use `Optional` here ? seems like a perfect use case.

Reply:
Replaced the code dealing with null here with `Optional` here https://github.com/googleapis/java-datastore/pull/823/commits/1584d9f88ae7af6a87223daaf5d0af6690e5a017.

Reply:
great!